### ISSUE TITLE
Bug: `No matching configuration found` warning for code block matched only by non-universal patterns

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#18493]
### Environment

Node version: v20.12.0
npm version: 10.5.0
Local ESLint version: v9.3.0
Global ESLint version: no
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

I wanted to make a repro with `eslint-plugin-markdown`, but since [this warning](https://github.com/eslint/eslint/blob/389744be255717c507fafc158746e579ac08d77e/lib/linter/linter.js#L1986-L1993) has `line: 0` (which might be another bug), `eslint-plugin-markdown` filters it out as out-of-range. So I made a simple processor that assumes that the entire markdown file is one ` ```ts ` code block.

```js
// eslint.config.js
export default [
  {
      plugins: {
          "my-plugin": {
              processors: {
                  "my-processor": {

                      // dummy processor that assumes that the whole content of a file is a single ```ts code block 
                      preprocess(text) {
                          const lines = text.split("\n");
                          return [{
                              filename: "foo.ts",
                              text: lines.slice(1, -1).join("\n")
                          }]
                      },

                      // just increment lines to adjust for the first ```ts line
                      postprocess(messages) {
                          const retv = messages[0]; // there's always exactly 1 code block

                          retv.forEach(message => {
                              message.line++
                              message.endLine++; 
                          });

                          return retv;
                      }
                  }
              }
          }
      }
  },
  {
      files: ["**/*.md"],
      processor: "my-plugin/my-processor"
  },
  {
      files: ["**/*"],
      rules: {
          "no-undef": 2
      }
  }
];

```

`test.md`:

````md
```ts
foo
```
````

### What did you expect to happen?

Since the `.ts` virtual file doesn't have an explicit match in the config, I'd expect it to be silently ignored, so at the end I'd expect no lint errors or warnings.

### What actually happened?

```
C:\projects\tmp\tmp\test.md
  1:0  warning  No matching configuration found for C:\projects\tmp\tmp\test.md\0_foo.ts

✖ 1 problem (0 errors, 1 warning)
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-hqxgng?file=eslint.config.js

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

We found this problem while discussing https://github.com/eslint/rewrite/pull/7.

I think we could just replace the following line with `return configs.getConfig(blockFilename) !== void 0;`

https://github.com/eslint/eslint/blob/389744be255717c507fafc158746e579ac08d77e/lib/eslint/eslint.js#L512

**Comment by fasttime** [comment]
Thanks for the issue and the repro. Marked as accepted.

**Comment by mdjermanovic** [comment]
I'm working on this.
